src/config.rs

```diff
@@ -0,0 +1,122 @@
+use clap::Parser;
+use num_rational::Ratio;
+use serde::Serialize;
+
+use crate::seat::Seat;
+
+#[derive(Parser, Serialize, Debug)]
+pub struct Config {
+    #[arg(long)]
+    pub num_blocks: u64,
+    #[arg(long)]
+    pub num_shards: u64,
```

Comment:
I doubt the number of shards will exceed `u32::MAX`, so it might make sense to declare this as as a `u32` instead. Then casting to `usize` shouldn't require `try_from` anymore, which will remove some unnecessary error handling.

Reply:
I agree that the number of shards most likely will not grow beyond `u32::MAX`, though I’m not sure if using `u32` avoids error handling. The largest unsigned integer with an implementation of `From` for `usize` is [`u16`](https://doc.rust-lang.org/std/primitive.usize.html#impl-From%3Cu16%3E-for-usize). Though perhaps I’ve missed another way that allows infallible conversion from `u32` to `usize`?

Reply:
I'll change it to `u16`, which is sufficient for our use case. If ever more than `u16::MAX` shards are needed, I think we can change it to `u32` without breaking existing CLI usage (any valid `--num-shards` parameter remains valid).